### PR TITLE
30 refactoring exercise

### PR DESCRIPTION
**Pull request by ErinGreenhalgh**
**What's this PR do?**
This PR submits my refactors to Block, Segment, and Pattern
- In Block, corrected global variables in for loops
- In Segment, remove unused code and whitespace
- In Pattern, refactor drawComputerBlocks function into two functions, one for placing and coloring the computer square and one for drawing it on the canvas; also added tests

**Where should the reviewer start?**
Work up from Block, to Segment, to Pattern

**How should this be manually tested?**
Run the server and check that the game functions in development: i.e. for one level, see the two computer dots, place your dot, watch them all fill, and progress to the next level if you win. 

**Any background context you want to provide?**
Pattern's drawComputerBlocks method randomly picks two blocks on the current pattern to place a computer block, so they will be rendered when the user starts the level. 

[**What gif best describes this PR or how it makes you feel?**](http://giphy.com/gifs/happy-sokka-exicted-Ofv6UfdjhRYdy)

@seeker105 @rrgayhart

### CODE REVIEW

**Comment by rrgayhart** [comment]
PR body filled out like a boss! Excellent!

File: lib/block.js [hunk 1]
```diff
@@ -11,8 +11,8 @@ function Block(options){
 
 Block.prototype.coordinates = function () {
   var coordinates = []
-  for(x = this.x; x <= (this.x + this.width); x++){
-    for(y = this.y; y <= (this.y + this.height); y++){
+  for( var x = this.x; x <= (this.x + this.width); x++){
```
Comment (rrgayhart) [inline lib/block.js:14]: 👍

File: lib/segment.js [hunk 2]
```diff
@@ -48,20 +48,11 @@ Segment.prototype.allBlocks = function(){
   return result;
 };
 
-Segment.prototype.colorAmount = function (color) {
```
Comment (rrgayhart) [inline lib/segment.js:51]: Was there a test for this that is now breaking? I'm assuming not - but just checking

File: lib/block.js (hunk 1 above)
Comment (seeker105) [reply lib/block.js:14]: Great change. Glad she went over that this morning, I didn't remember the importance of this.

File: lib/pattern.js [hunk 3]
```diff
@@ -46,20 +46,25 @@ function drawAllBlocks(blocks, compColors, context){
 }
 
 function drawComputerBlock(blocks, compColors, context){
-  var compBlocks = compColors.map(function(color) {
-    var block = blocks[Math.floor(Math.random() * blocks.length)];
-    block.color = color;
-    return block;
-  });
+  var compBlocks = randomlyPlaceAndColorCompBlocks(blocks, compColors);
```
Comment (seeker105) [inline lib/pattern.js:49]: Good idea to make this a separate function.

**Pull request merged**